src/js/pouch.js

```diff
@@ -25,42 +46,93 @@ gpii.pouch.init = function (that) {
-        if (dbConfig.data) {
-            var data = require(dbConfig.data);
-            db.bulkDocs(data);
-        }
-    });
-
     that.expressPouchdb = expressPouchdb(MemPouchDB, { configPath: pouchConfigPath });
 
-    //that.expressPouchdb.couchLogger.setFile(logPath);
+    var waitForCleaning = PouchDB.isBeingCleaned ? PouchDB.isBeingCleaned : fluid.promise();
+    waitForCleaning.then(function () {
+        var promises = [];
+        fluid.each(that.options.databases, function (dbConfig, key) {
+            var db = new MemPouchDB(key);
+            that.databaseInstances[key] = db;
+            if (dbConfig.data) {
+                var data = require(dbConfig.data);
+                promises.push(db.bulkDocs(data));
+            }
+        });
+
+        when.all(promises).then(function () {
+            that.events.onStarted.fire();
+        });
+    });
 
-    that.events.onStarted.fire();
+    if (PouchDB.isBeingCleaned) {
+        fluid.log("Waiting for the last run to finish its cleanup...");
+    }
+    else {
+        fluid.log("No previous run detected. Continuing with the normal startup...");
+        waitForCleaning.resolve({});
+    }
 };
 
 gpii.pouch.getRouter = function (that) {
     return that.expressPouchdb;
 };
 
+// Remove all data from each database between runs, otherwise we encounter problems with data leaking between tests.
+//
+// https://github.com/pouchdb/pouchdb/issues/4124
+//
+gpii.pouch.cleanup = function (that) {
+    var promises = [];
+    fluid.each(that.databaseInstances, function (db, key) {
+            // If we use the simpler method, the next attempt to recreate the database fails with a 409 document update conflict.
+            //var promise = db.destroy();
 
-// TODO:  Write a change listener to allow easy adding of new databases
+            // Instead, we retrieve the list of all document IDs and revisions, and then bulk delete them.
+            var promise = db.allDocs()
+                .then(function (result) {
+                    var bulkPayloadDocs = fluid.transform(result.rows, gpii.pouch.transformRecord);
+                    var bulkPayload     = {docs: bulkPayloadDocs};
+                    return db.bulkDocs(bulkPayload);
+                })
+                .then(function () {
+                    fluid.log("Deleted existing data from database '" + key + "'...");
+                    return db.compact();
+                })
+                .then(function () {
+                    fluid.log("Compacted existing database '" + key + "'...");
+                })
+                .catch(fluid.fail); // jshint ignore:line
 
-/*
-    The "databases" option is expected to be an array keyed by dbName, with options to control whether data is loaded or not, as in:
+            promises.push(promise);
+        }
+    );
+
+    // Make sure that the next instance of pouch knows to wait for us to finish cleaning up.
+    PouchDB.isBeingCleaned = when.all(promises);
```

Comment:
To head off similar problems in future, add a hard failure here if PouchDB.isBeingCleaned currently holds a value - note that your current implementation would only be good for 2 tests since you had not performed any cleanup of the cleanup marker :)